x/committee/keeper/proposal.go

```diff
@@ -0,0 +1,162 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/kava-labs/kava/x/committee/types"
+)
+
+// SubmitProposal adds a proposal to a committee so that it can be voted on.
+func (k Keeper) SubmitProposal(ctx sdk.Context, proposer sdk.AccAddress, committeeID uint64, pubProposal types.PubProposal) (uint64, sdk.Error) {
+	// Limit proposals to only be submitted by committee members
+	com, found := k.GetCommittee(ctx, committeeID)
+	if !found {
+		return 0, types.ErrUnknownCommittee(k.codespace, committeeID)
+	}
+	if !com.HasMember(proposer) {
+		return 0, sdk.ErrUnauthorized("proposer not member of committee")
+	}
+
+	// Check committee has permissions to enact proposal.
+	if !com.HasPermissionsFor(pubProposal) {
+		return 0, sdk.ErrUnauthorized("committee does not have permissions to enact proposal")
+	}
+
+	// Check proposal is valid
+	if err := k.ValidatePubProposal(ctx, pubProposal); err != nil {
+		return 0, err
+	}
+
+	// Get a new ID and store the proposal
+	deadline := ctx.BlockTime().Add(com.MaxProposalDuration)
```

Comment:
Each proposal before a committee is subject to constant duration `MaxProposalDuration`. Consider renaming to `ProposalDuration` since proposal duration times don't vary by committee.